eth_utils/conversions.py

```diff
@@ -19,33 +26,37 @@
 )
 
 
+HexStr = NewType('HexStr', str)
+Primitives = Union[bytes, int, bool]
+T = TypeVar('T')
+
+
 @validate_conversion_arguments
-def to_hex(primitive=None, hexstr=None, text=None):
+def to_hex(primitive: Primitives=None, hexstr: HexStr=None, text: str=None) -> HexStr:
     """
     Auto converts any supported value into its hex representation.
-
     Trims leading zeros, as defined in:
     https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding
     """
     if hexstr is not None:
-        return add_0x_prefix(hexstr.lower())
+        return HexStr(add_0x_prefix(hexstr.lower()))
```

Comment:
We should be able to remove this cast if `add_0x_prefix` is first updated to return a `HexStr` which I think it should?

Reply:
I'm not sure that's the move - updating `add_0x_prefix` (and by the same logic also `remove_0x_prefix`) to return `HexStr`s basically triggers a whole lot more casting that would need to be done to handle it